src/include/sof/audio/kpb.h

```diff
@@ -33,21 +33,22 @@ struct comp_buffer;
 
 /* KPB internal defines */
 #define KPB_MAX_BUFF_TIME 2100 /**< time of buffering in miliseconds */
-#define KPB_MAX_SUPPORTED_CHANNELS 2
-#define	KPB_SAMPLING_WIDTH 16 /**< number of bits */
-#define	KPB_SAMPLNG_FREQUENCY 16000 /* max sampling frequency in Hz */
-#define KPB_NR_OF_CHANNELS 2
+#define KPB_MAX_SUPPORTED_CHANNELS 2 /**< number of supported channels */
+/**< number of samples taken each milisecond */
+#define	KPB_SAMPLES_PER_MS (KPB_SAMPLNG_FREQUENCY / 1000)
```

Comment:
@mrajwa Could you put this patch as first in the series? This would make the rest easier to follow with correct terms used. 

And I think KPB_SAMPLING_WIDTH was not mislabelled. E.g. in kpb_is_sample_width_supported() same term is used to refer to bit depth of a single sample (with 16bit and 24bit supported). Nobody is using this, so maybe you can just remove this.

So in the end, I don't think you should add KPB_SAMPLES_PER_MS either. The kpb.c code is a bit mix with some parts of code taking sampling-freq from config struct, and other places using KPB_SAMPLING_FREQUENCY. To make it more confusing this header documents the constant as "max sampling frequency" while it used as the only supported freq.

I think it's clearer you calculate bytes_per_ms in the implementation side direct (maybe in init), using kpb->config.sampling_freq. If the component is later udpated to supported more than one frequency, then it's clearer. If you prefer to only support one freq, update the document to reflect this and add KPB_SAMPLES_PER_MS in the patch that uses it.
I think you shouldn't

Reply:
> Nobody is using this, so maybe you can just remove this.

For sure kpb_is_sample_width_supported() has its use. Check kpb_new() function. Also, imho there sampling_width has correct meaning. 



> To make it more confusing this header documents the constant as "max sampling frequency" while it used as the only supported freq.

That's right, I have just changed the comment message to clearly state it is the only supported frequency.

Reply:
>> Nobody is using this, so maybe you can just remove this. 
> For sure kpb_is_sample_width_supported() has its use. Check kpb_new() function. 

I meant KPB_SAMPLING_WIDTH macro is not used by anyone else than kpb_init_draining(). 

> That's right, I have just changed the comment message to clearly state it is the only supported frequency.

Ack, thanks.